### PR TITLE
fix: updated ingredients_categories_and_types

### LINKED ISSUE #8890: Piments (vert, rouge, jaune) is not recognized by ingredients parsing
Alizarine reported:
image

I have produced a quick and dirty [Mirabelle request](https://mirabelle.openfoodfacts.org/products?sql=select+code%2C+url%0D%0A%2C+ingredients_text%0D%0A--%2C+ingredients_tags%2C+ingredients_analysis_tags%0D%0A%2C+regexp_substr%28ingredients_text%2C+%22%5CS%2B+%3F%5C%28%5B%5E%5C%29%5D*%2C+%3F%5B%5E%5C%29%5D*%5C%29%22%29+as+ingr_with_parenthesis%0D%0A%2C+count%28regexp_substr%28ingredients_text%2C+%22%5CS%2B+%3F%5C%28%5B%5E%5C%29%5D*%2C+%3F%5B%5E%5C%29%5D*%5C%29%22%29%29+as+nb%0D%0Afrom+%5Ball%5D%0D%0Awhere+ingredients_text+REGEXP+%22%5CS%2B+%3F%5C%28%5B%5E%5C%29%5D*%2C+%5B%5E%5C%29%5D*%5C%29%22%0D%0Agroup+by+ingr_with_parenthesis%0D%0Aorder+by+nb+desc%0D%0Alimit+30000) to see if there are many cases like this but it appears to be very rare.
It might be fixed with hard-coded rules.

### Part of
- #5706

### PR DESCRIPTION
**Pull request by benbenben2**
### What
Added piment in %ingredients_categories_and_types in Ingredients.pm.
Because it resulted to "piments de vert, piments de rouge, piments de jaune", a new variable (boolean) was added that can be set to 0 to indicate that there is no additional word ("de" or "d' ") between the categories (piment, for example) and the type (vert, for example).

### Screenshot
Screenshot_20230919_195258

### Related issue(s) and discussion
- Fixes #8890

### CODE REVIEW

File: lib/ProductOpener/Ingredients.pm [hunk 1]
```diff
@@ -4719,6 +4721,8 @@ my %ingredients_categories_and_types = (
 				"manganèse", "potassium", "sodium", "zinc",
 			]
 		],
+		# peppers
+		[["piment",], ["vert", "jaune", "rouge",], 0,],
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm:4725]: There are a few products with "poivrons (rouge, jaune, vert)" like https://fr.openfoodfacts.org/produit/8001060035829/trio-de-poivrons-aubergines-sacla

```suggestion
		[["piment", "poivron"], ["vert", "jaune", "rouge",], 0,],
```

File: lib/ProductOpener/Ingredients.pm [hunk 2]
```diff
@@ -4792,6 +4796,7 @@ sub develop_ingredients_categories_and_types ($ingredients_lc, $text) {
 
 	if (defined $ingredients_categories_and_types{$ingredients_lc}) {
 
+		# my $categories_and_types_ref_index = 0;
```
Comment (stephanegigandet) [inline lib/ProductOpener/Ingredients.pm:4799]: should this comment be there?

**Review by stephanegigandet — COMMENTED** [review]
Looks great! Just one minor suggestion to add "poivron" too

File: lib/ProductOpener/Ingredients.pm (hunk 2 above)
Comment (benbenben2) [reply lib/ProductOpener/Ingredients.pm:4799]: nop, I will remove it

**Review by stephanegigandet — APPROVED** [review]
Thank you very much!

**Pull request merged**